### PR TITLE
feat: add rancher-terraform and import-existing-vm

### PR DESCRIPTION
**Pull request by FrankYang0529**
https://github.com/harvester/harvester/issues/6147

### CODE REVIEW

File: docs/rancher/rancher-terraform.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+---
+
+[Rancher Terraform](https://registry.terraform.io/providers/rancher/rancher2/) is a terraform provider that allows administrators to create and manage RKE2 guest clusters using Terraform.
+
+## Deploying
+
+### Prerequisites
+
+- The Kubernetes cluster is built on top of Harvester virtual machines.
+- The Harvester virtual machines that run as guest Kubernetes nodes are in the same namespace.
```
Comment (jillian-maroket) [inline docs/rancher/rancher-terraform.md:22]:
```suggestion
- The Kubernetes cluster is built on top of Harvester VMs.
- The Harvester VMs that run as guest Kubernetes nodes are in the same namespace.
```

File: docs/rancher/rancher-terraform.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+
+### Deploying Guest Cluster with Rancher Terraform
+
+1. Create an API key on Rancher dashboard. Account & API Keys > Create API key > Create.
```
Comment (jillian-maroket) [inline docs/rancher/rancher-terraform.md:27]:
```suggestion
### Deploy Guest Clusters Using the Rancher Terraform Provider

1. Create an API key. 

    On the Rancher UI, go to **Account & API Keys** > **Create API key** > **Create**.
```

File: docs/rancher/rancher-terraform.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+    ![Create API Key](/img/v1.4/rancher/create-api-key.png)
+    ![Access & Secret Keys](/img/v1.4/rancher/access-and-secret-keys.png)
+
+1. Get the Harvester cluster ID. Virtualization Management > Manage > Related Resources > Mgmt Cluster Name.
```
Comment (jillian-maroket) [inline docs/rancher/rancher-terraform.md:32]:
```suggestion
1. Obtain the Harvester cluster ID. 

    On the Rancher UI, go to **Virtualization Management** > **Manage** > **Related Resources** > **Mgmt Cluster Name**.
```

File: docs/rancher/rancher-terraform.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+    ![Harvester Cluster ID](/img/v1.4/rancher/harvester-cluster-id.png)
+
+1. Get kubeconfig for Harvester Cloud Provider and Harvester CSI Driver.
```
Comment (jillian-maroket) [inline docs/rancher/rancher-terraform.md:36]:
```suggestion
1. Obtain the kubeconfig for the Harvester Cloud Provider and the Harvester CSI Driver.

```

File: docs/rancher/rancher-terraform.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+            harvester-cloud-provider:
+            clusterName: rke2-demo
+            cloudConfigPath: /var/lib/rancher/rke2/etc/config-files/cloud-provider-config
+            EOF
+        }
+    }
+    ```
+
+1. Run `terraform init`.
+1. Run `terraform apply`
```
Comment (jillian-maroket) [inline docs/rancher/rancher-terraform.md:182]:
```suggestion
1. Run `terraform init`.

1. Run `terraform apply`.
```

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

**Review by jillian-maroket — APPROVED** [review]
LGTM. Thanks!

File: docs/rancher/rancher-terraform.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+            package_update: true
+            packages:
+            - qemu-guest-agent
+            - iptables
+            runcmd:
+            - - systemctl
+                - enable
+                - '--now'
+                - qemu-guest-agent.service
+            password: test
```
Comment (bk201) [inline docs/rancher/rancher-terraform.md:129]: Let's remove this from the example, it's good for debugging purposes but not for production.

**Review by bk201 — APPROVED** [review]
@khushboo-rancher Can you check the doc PR? Thanks.

File: docs/rancher/rancher-terraform.md (hunk 6 above)
Comment (FrankYang0529) [reply docs/rancher/rancher-terraform.md]: Yes, removed it. Thanks.

File: docs/rancher/csi-driver.md [hunk 7]
```diff
@@ -276,3 +276,29 @@ Now you can create a new StorageClass that you intend to use in your guest Kuber
     :::
 
 1. You can now create a PVC based on this new **StorageClass**, which utilizes the **Host StorageClass** to provision volumes on the bare-metal Harvester cluster.
+
+## Upgrade the CSI Driver
+
+### Upgrade RKE2
+
+You can upgrade the CSI driver by upgrading RKE2 using the Rancher UI.
```
Comment (khushboo-rancher) [inline docs/rancher/csi-driver.md:284]: Can we write description something like below
"To upgrade the CSI driver, use the Rancher UI to upgrade RKE2. Ensure the new RKE2 version supports/bundled with the updated CSI driver version."

As, simply upgrading RKE2 will not upgrade the CSI driver. WDYT?

File: docs/rancher/import-existing-vm.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
+---
+sidebar_position: 8
+sidebar_label: Import Existing VM
+title: "Import Existing VM"
```
Comment (khushboo-rancher) [inline docs/rancher/import-existing-vm.md:4]: Should this be "Import Existing Cluster built on Harvester VM"?

**Review by khushboo-rancher — COMMENTED** [review]
This is great! just two minor comments.
Thanks @FrankYang0529

File: docs/rancher/import-existing-vm.md (hunk 8 above)
Comment (FrankYang0529) [reply docs/rancher/import-existing-vm.md]: Yes, that is better. Updated it!

File: docs/rancher/csi-driver.md (hunk 7 above)
Comment (FrankYang0529) [reply docs/rancher/csi-driver.md]: Agree, updated it. Thanks for the suggestion.

File: versioned_docs/version-v1.3/rancher/csi-driver.md [hunk 9]
```diff
@@ -279,5 +279,21 @@ Now you can create a new StorageClass that you intend to use in your guest Kuber
...
-  
-1. You can now create a PVC based on this new **StorageClass**, which utilizes the **Host StorageClass** to provision volumes on the bare-metal Harvester cluster.
\ No newline at end of file
+
+1. You can now create a PVC based on this new **StorageClass**, which utilizes the **Host StorageClass** to provision volumes on the bare-metal Harvester cluster.
+
+## Upgrade CSI Driver
+
+### Upgrade RKE2
+The csi driver can be upgraded by upgrading the RKE2 version. You can upgrade the RKE2 cluster via the Rancher UI as follows:
```
Comment (khushboo-rancher) [inline versioned_docs/version-v1.3/rancher/csi-driver.md:288]: Looks like, the statement is missed to be modified at the version level doc.
```
To upgrade the CSI driver, use the Rancher UI to upgrade RKE2. Ensure the new RKE2 version supports/bundled with the updated CSI driver version.
```

Comment (FrankYang0529) [reply versioned_docs/version-v1.3/rancher/csi-driver.md]: Thank you. Updated it.

**Review by khushboo-rancher — APPROVED** [review]

**Pull request merged**